starfish/models/surfer_model.py

```diff
@@ -154,3 +173,28 @@ def set_http_client(http_client):
         """Set the http client to something other than the default `requests`"""
         SurferModel._http_client = http_client
 
+    @staticmethod
+    def get_authorization_token(surfer_url, surfer_username, surfer_password):
+        """Get a surfer authorization token (create one if needed).
+        Throws exception on error."""
+        token_url = surfer_url + SURFER_BASE_URI + '/auth/token'
+        r = requests.get(token_url, auth=(surfer_username, surfer_password))
+        token = None
+        if r.status_code == 200:
+            tokens = r.json()
+            if len(tokens) > 0:
+                token = tokens[-1]
+            else: # need to create a token
+                r = requests.post(token_url, auth=(surfer_username, surfer_password))
+                if r.status_code == 200:
+                    token = r.json()
+                else:
+                    msg = f'unable to create token, status {r.status_code}'
```

Comment:
Are there conditions other that this:(user does not have permissions to get a token) , which may cause this error?

Reply:
Sure.. all kinds of things can go wrong... The server may be down or throw a 500 or have an invalid TLS certificate.. We _must_ check the return status of all web service calls.